test/bulkstatus.service.test.js

```diff
@@ -0,0 +1,59 @@
+const { bulkStatusSetup, cleanUpDb } = require('./populateTestData');
+const { buildConfig } = require('../src/util/config');
+const { initialize } = require('../src/server/server');
+const config = buildConfig();
+const server = initialize(config);
+const supertest = require('supertest');
+
+describe('checkBulkStatus logic', () => {
+  beforeAll(bulkStatusSetup);
+  test('check 202 returned for pending request', async () => {
+    await supertest(server.app)
+      .get('/4_0_0/bulkstatus/PENDING_REQUEST')
+      .expect(202)
+      .then(response => {
+        expect(response.headers['x-progress']).toEqual('Retrieving export files');
+        expect(response.headers['retry-after']).toEqual('120');
+      });
+  });
+  test('check 200 returned for completed request', async () => {
+    await supertest(server.app)
+      .get('/4_0_0/bulkstatus/COMPLETED_REQUEST')
+      .expect(200)
+      .then(response => {
+        expect(response.headers.expires).toBeDefined();
+        expect(response.headers['content-type']).toEqual('application/json; charset=utf-8');
+        expect(response.body).toBeDefined();
+      });
+  });
+  test('check 500 and error returned for failed request with known error', async () => {
+    await supertest(server.app)
+      .get('/4_0_0/bulkstatus/KNOWN_ERROR_REQUEST')
+      .expect(500)
+      .then(response => {
+        expect(response.body.issue[0].code).toEqual('ErrorCode');
+        expect(response.body.issue[0].details.text).toEqual('Known Error Occurred!');
```

Comment:
Based on [this documentation](https://github.com/smart-on-fhir/bulk-import/blob/master/import-pnp.md#bulk-data-import-status-request-data-provider-to-data-consumer) , do we need to check that the content-type is application/json+fhir when we get a 400 or 500 status code?

Reply:
might as well. Good call